.github/workflows/publish_osdk_and_ostd.yml

```diff
@@ -0,0 +1,69 @@
+name: Publish OSDK and OSTD
+
+on:
+  pull_request:
+    paths:
+      - VERSION
+      - ostd/**
+      - osdk/**
+  push:
+    branches:
+      - main
+    paths: 
+      - VERSION
+
+jobs:
+  osdk-publish:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    container: asterinas/asterinas:0.6.2
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Check Publish OSDK
+        # On pull request, set `--dry-run` to check whether OSDK can publish
+        if: github.event_name == 'pull_request'
+        run: |
+          cd osdk
+          cargo publish --dry-run
+        
+      - name: Publish OSDK
+        # On push, OSDK will be published
+        if: github.event_name == 'push'
```

Comment:
Did I miss something? Why are we going to publish OSDK and OSTD on every update?

Reply:
The push event will only be triggered when the files listed in the `paths` are changed. Currently, the `paths` only contains VERSION.
```yaml
on:
  push:
    branches:
      - main
    paths: 
      - VERSION
```